pycontrails/models/cocip/wake_vortex.py

```diff
@@ -373,3 +397,378 @@ def initial_contrail_depth(
+        provided as a length scale equivalent, [:math:`m`]
+
+    Notes
+    -----
+    - See eq. (7) in :cite:`unterstrasserPropertiesYoungContrails2016`.
+    """
+    # Solve non-linear equation numerically using the bisection method
+    # Did not use scipy functions because it is unstable when dealing with np.arrays
+    z_1 = np.zeros_like(rho_emit)
+    z_2 = np.full_like(rho_emit, 1000.0)
+
+    lhs = (thermo.e_sat_ice(air_temperature) / (constants.R_v * air_temperature)) + rho_emit
+
+    for _ in range(n_iter):
+        z_est = 0.5 * (z_1 + z_2)
+        rhs = thermo.e_sat_ice(air_temperature + 9.8e-3 * z_est) / (
+            constants.R_v * (air_temperature + 9.8e-3 * z_est)
+        )
+        z_1[lhs > rhs] = z_est[lhs > rhs]
+        z_2[lhs < rhs] = z_est[lhs < rhs]
+
+    return 0.5 * (z_1 + z_2)
+
+
+def plume_area(wingspan: npt.NDArray[np.float64] | float) -> npt.NDArray[np.float64] | float:
+    """Calculate area of the wake-vortex plume.
+
+    Parameters
+    ----------
+    wingspan : npt.NDArray[np.float64] | float
+        aircraft wingspan, [:math:`m`]
+
+    Returns
+    -------
+    float
+        Area of two wake-vortex plumes, [:math:`m^{2}`]
+
+    Notes
+    -----
+    - See eq. (A6) and (A7) in :cite:`unterstrasserPropertiesYoungContrails2016`.
+    """
+    r_plume = 1.5 + 0.314 * wingspan
+    return 2.0 * 2.0 * np.pi * r_plume**2
+
+
+def z_desc_length_scale(
+    wingspan: npt.NDArray[np.float64] | float,
+    air_temperature: npt.NDArray[np.float64],
+    air_pressure: npt.NDArray[np.float64],
+    true_airspeed: npt.NDArray[np.float64],
+    aircraft_mass: npt.NDArray[np.float64],
+    dT_dz: npt.NDArray[np.float64],
+) -> npt.NDArray[np.float64]:
+    """Calculate the final vertical displacement of the wake vortex.
+
+    Parameters
+    ----------
+    wingspan : npt.NDArray[np.float64] | float
+        aircraft wingspan, [:math:`m`]
+    air_temperature : npt.NDArray[np.float64]
+        ambient temperature for each waypoint, [:math:`K`]
+    air_pressure : npt.NDArray[np.float64]
+        pressure altitude at each waypoint, [:math:`Pa`]
+    true_airspeed : npt.NDArray[np.float64]
+        true airspeed for each waypoint, [:math:`m s^{-1}`]
+    aircraft_mass : npt.NDArray[np.float64] | float
+        aircraft mass for each waypoint, [:math:`kg`]
+    dT_dz : npt.NDArray[np.float64]
+        potential temperature gradient, [:math:`K m^{-1}`]
+
+    Returns
+    -------
+    npt.NDArray[np.float64]
+        Final vertical displacement of the wake vortex, [:math:`m`]
+
+    Notes
+    -----
+    - See eq. (4) in :cite:`unterstrasserPropertiesYoungContrails2016`.
+    """
+    gamma_0 = _initial_wake_vortex_circulation(
+        wingspan, air_temperature, air_pressure, true_airspeed, aircraft_mass
+    )
+    n_bv = thermo.brunt_vaisala_frequency(air_pressure, air_temperature, dT_dz)
+    return ((8.0 * gamma_0) / (np.pi * n_bv)) ** 0.5
+
+
+def _initial_wake_vortex_circulation(
```

Comment:
I marked this function as non-public because it is currently not used elsewhere apart from the z_desc_length_scale` function. I don't have any strong preferences on this matter.